### PR TITLE
 MDK Mission tutorials- level up with MDK

### PR DESCRIPTION
**Pull request by jitendrakansal**
updated MDK Mission tutorials- level up with MDK
https://developers.sap.com/mission.mobile-dev-kit-level-up.html

### CODE REVIEW

**Comment by jitendrakansal** [comment]
updated tutorials for other two missions as well:

- https://developers.sap.com/mission.mobile-dev-kit-brand.html
- https://developers.sap.com/mission.mobile-dev-kit-controls.html

File: tutorials/cp-mobile-dev-kit-error-archive/cp-mobile-dev-kit-error-archive.md [hunk 1]
```diff
@@ -49,65 +49,59 @@ This step includes creating the mobile development kit project in the editor.
...
+6. In *OData Collections* step, unselect `Customers`, select `PurchaseOrderHeaders` and `PurchaseOrderItems`.
 
-    Click **Next** to finish the project creation.
+    Click **Finish** to complete the project creation.
 
-    !![MDK](img_1.8.png)
+    !![MDK](img_1.6.png)
 
-9. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_ErrorArchive` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+7. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_ErrorArchive` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-error-archive/cp-mobile-dev-kit-error-archive.md:102]:
```suggestion
7. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_ErrorArchive` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-map-extension/cp-mobile-dev-kit-map-extension.md [hunk 2]
```diff
@@ -43,57 +38,51 @@ This step includes creating the mobile development kit project in the editor.
...
     Since you will create an online based app, hence _Enable Offline Store_ option is unchecked.
 
-8. In *OData Collections* step, select `Customers` (if not selected by default). Click **Next** to finish the project creation.
+6. In *OData Collections* step, select `Customers` (if not selected by default). Click **Next** to finish the project creation.
 
-    !![MDK](img_1.8.png)    
+    !![MDK](img_1.6.png)    
 
-9. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Maps` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+7. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Maps` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-map-extension/cp-mobile-dev-kit-map-extension.md:83]:
```suggestion
7. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Maps` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-nativescript-geolocation/cp-mobile-dev-kit-nativescript-geolocation.md [hunk 3]
```diff
@@ -35,35 +35,29 @@ In this tutorial, you will use the existing NativeScript plugin nativescript-geo
...
+    >The _MDK Empty Project_ template creates a Logout action, Close page action, rule and an empty page (`Main.page`). After using this template, you can focus on creating your pages, other actions, and rules needed for your application. More details on _MDK template_ is available in [help documentation](https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/getting-started/mdk/webide.html#creating-a-new-project).
 
-6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Geolocation` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Geolocation` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-nativescript-geolocation/cp-mobile-dev-kit-nativescript-geolocation.md:58]:
```suggestion
5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Geolocation` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-online-app/cp-mobile-dev-kit-online-app.md [hunk 4]
```diff
@@ -35,61 +35,51 @@ This step includes creating the mobile development kit project in the editor.
...
 
     >Regardless of whether you are creating an online or offline application, this step is needed app to connect to an OData service. When building an Mobile Development Kit application, it assumes the OData service created and the destination that points to this service is setup in Mobile Services.
 
-8. In **OData Collections** step, select `Customers`, `Product`, `SalesOrderHeaders` and `SaelesOrderItems` data collections. Click **Finish**.
-
-    !![MDK](img_2.8.png)
-
-9. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `DemoSampleApp` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDKOnlineApp` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-online-app/cp-mobile-dev-kit-online-app.md:80]:
```suggestion
6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDKOnlineApp` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-push/cp-mobile-dev-kit-push.md [hunk 5]
```diff
@@ -49,33 +51,31 @@ You may clone an existing metadata project from [GitHub repository](https://gith
...
 
     >More details on _MDK template_ is available in [help documentation](https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/getting-started/mdk/bas.html#creating-a-new-project-cloud-foundry).  
 
-6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Push` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+    >If you see *Cloud foundry token expired, continue without mobile services connection?* message, then set the Cloud Foundry environment again by clicking at bottom left corner of your status bar to initiate a valid session and repeat above steps.      
+
+5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Push` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-push/cp-mobile-dev-kit-push.md:76]:
```suggestion
5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Push` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-rest-api/cp-mobile-dev-kit-rest-api.md [hunk 6]
```diff
@@ -99,94 +95,73 @@ With above details, you will next configure an app in Mobile Services, add root
...
 
     >More details on _MDK template_ is available in [help documentation](https://help.sap.com/doc/f53c64b93e5140918d676b927a3cd65b/Cloud/en-US/docs-en/guides/getting-started/mdk/bas.html#creating-a-new-project-cloud-foundry).
 
-6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Petstore` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+    >If you see *Cloud foundry token expired, continue without mobile services connection?* message, then set the Cloud Foundry environment again by clicking at bottom left corner of your status bar to initiate a valid session and repeat above steps.      
+
+5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Petstore` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-rest-api/cp-mobile-dev-kit-rest-api.md:127]:
```suggestion
5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Petstore` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-rest-api/cp-mobile-dev-kit-rest-api.md [hunk 7]
```diff
@@ -398,92 +355,63 @@ In this step, you will create the `Pet_Create.page` as a **Form Cell Page**. Thi
...
+    | Field | Value |
     |----|----|
-    | `Name`| `FCCreateStatus` |
-    | `Caption` | `Status` |
-    | `IsEditable`| Select `true` from the dropdown |
-    | `PlaceHolder`| `Enter Value` |
+    | `Method`| `POST` |
 
-    !![MDK](img_10.9.png)    
+4. Under `Body`, switch to `object type` by clicking on the icon, once it's color has changed click on `Body[0]` to add array items, this should now display a create icon in front of `Body[0]`. Click Create icon to create an array item(0).
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-rest-api/cp-mobile-dev-kit-rest-api.md:392]:
```suggestion
4. Under `Body`, switch to `object type` by clicking the icon, once it's color has changed click `Body[0]` to add array items, this should now display a create icon in front of `Body[0]`. Click Create icon to create an array item(0).
```

File: tutorials/cp-mobile-dev-kit-slider-extension/cp-mobile-dev-kit-slider-extension.md [hunk 8]
```diff
@@ -25,11 +25,7 @@ You may clone an existing project from [GitHub repository](https://github.com/SA
 ---
 
 
-To extend the functionality, or customize the look and feel, and behavior of your client app, you can create extension controls other than the already existing MDK built-in controls using the following:
-
--  NativeScript (TypeScript/JavaScript applicable for both Android and iOS)
-
--  Swift class (iOS only)
+To extend the functionality, or customize the look and feel, and behavior of your client app, you can create extension controls other than the already existing MDK built-in controls using **NativeScript** (TypeScript/JavaScript applicable for both Android and iOS)
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-slider-extension/cp-mobile-dev-kit-slider-extension.md:28]:
```suggestion
To extend the functionality, or customize the look and feel, and behavior of your client app, you can create extension controls other than the already existing MDK built-in controls by using **NativeScript** (TypeScript/JavaScript applicable for both Android and iOS)
```

File: tutorials/cp-mobile-dev-kit-slider-extension/cp-mobile-dev-kit-slider-extension.md [hunk 9]
```diff
@@ -40,33 +36,31 @@ In this tutorial, you will create a Slider extension via NativeScript (in TypeSc
...
 
-6. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Petstore` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+    >If you see *Cloud foundry token expired, continue without mobile services connection?* message, then set the Cloud Foundry environment again by clicking at bottom left corner of your status bar to initiate a valid session and repeat above steps.   
 
-    !![MDK](img_1.6.png)
+5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Slider` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-slider-extension/cp-mobile-dev-kit-slider-extension.md:61]:
```suggestion
5. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Slider` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

File: tutorials/cp-mobile-dev-kit-style/cp-mobile-dev-kit-style.md [hunk 10]
```diff
@@ -84,13 +78,13 @@ This step includes creating the mobile development kit project in the editor.
 
     Since you will create an offline based app, hence **Enable Offline** option is selected.
 
-8. In *OData Collections* step, select `Customers` (if not selected as default). Click **Next** to finish the project creation.
+8. In *OData Collections* step, select `Customers` (if not selected as default). Click **Finish** to complete the project creation.
 
     !![MDK](img_1.8.png)
 
-9. After clicking **Next**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Styling` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace or to add it to workspace. Ignore the pop-up or click the cross icon to hide the window.
+9. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Styling` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in new workspace. Ignore the pop-up or click the cross icon to hide the window.
```
Comment (MichaelCzcz) [inline tutorials/cp-mobile-dev-kit-style/cp-mobile-dev-kit-style.md:85]:
```suggestion
9. After clicking **Finish**, the wizard will generate your MDK Application based on your selections. You should now see the `MDK_Styling` project in the project explorer. As you have already opened the workspace, there is no need to open the generated project in a new workspace. Ignore the pop-up or click the cross icon to hide the window.
```

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]
There are some "missing words" errors and small style issues. For your convenience, I have made suggestions on how it should look like. You can either make the changes in your fork and push them to production or you can click "Commit suggestion", but you will have to apply them manually to QA.

**Comment by jitendrakansal** [comment]
@MichaelCzcz thanks for reviewing, i'have updated the tutorials with your feedback.

**Comment by MichaelCzcz** [comment]
Please check the links in all tutorials that are part of this mission and change them to the correct formatting where necessary.
Please also check the formatting, for example, spacing and indentation for images, of all tutorials and adjust it where necessary. For details please see the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9?_lightbox=true#mcetoc_1dtqfqchq1u).

**Comment by jitendrakansal** [comment]
@MichaelCzcz I didn't follow you completely.
Could you point me to one tutorial which does not have link in correct format so that i can update it in others?

**Comment by jitendrakansal** [comment]
@MichaelCzcz i have updated the PR with changes as we discussed in the call. Please check now.

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]
I have found more cases where the links are not formatted correctly. Also, please don't link to tutorial steps. Rather link to the tutorial and name the step in the text. Please make corrections.
https://github.com/SAPDocuments/Tutorials/blame/a6f757e0f01115c5887678b2fca87d14aae5eea4/tutorials/cp-mobile-dev-kit-annotations/cp-mobile-dev-kit-annotations.md#L45
https://github.com/SAPDocuments/Tutorials/blame/a6f757e0f01115c5887678b2fca87d14aae5eea4/tutorials/cp-mobile-dev-kit-build-client/cp-mobile-dev-kit-build-client.md#L185
https://github.com/SAPDocuments/Tutorials/blame/a6f757e0f01115c5887678b2fca87d14aae5eea4/tutorials/cp-mobile-dev-kit-build-client/cp-mobile-dev-kit-build-client.md#L257
https://github.com/SAPDocuments/Tutorials/blame/a6f757e0f01115c5887678b2fca87d14aae5eea4/tutorials/cp-mobile-dev-kit-link-entity/cp-mobile-dev-kit-link-entity.md#L23

**Comment by jitendrakansal** [comment]
@MichaelCzcz i have pushed the changes as per your last comment, please check now.

**Pull request merged**